client/src/components/EditBox.js

```diff
@@ -12,10 +12,26 @@ import {
 
 export default function EditBox() {
   let { id } = useParams();
+  console.log(id);
   const skills = fakeData[id];
   console.log(skills.length);
   const [deleted, setDeleted] = useState(skills);
   const [learningObjective, setLearningObjective] = useState(skills);
+  const [learningObj, setLearningObj] = useState([])
+  useEffect(() => {
+
+    fetch(`/api/learningobjectives/react`)
+    .then((response) => response.json())
+            .then((data) => {
+              if (data.error) {
+                throw data;
+              }
+              console.log(data);
+              setLearningObj(data)
+            })
+          },[])
```

Comment:
Remember the second argument to `useEffect`, after the callback function, is the dependencies list.
The callback function **is only ever called** when one or more dependencies changes. Here, you've hard coded an empty list `[]`, which means it will only run once (the first time the component renders) and will never run again, for example when the id changes!

You could fix this by making the `id` a dependency, that way when the `id` param changes, between `react`, `css`, `html` etc then this fetch callback will be called again.
```suggestion
          }, [id])
```

Reply:
@petemcfarlane  we did try `${id}` but we didn't add it to dependency array . thanks for getting back.